test/e2e/networking.go

```diff
@@ -102,160 +103,245 @@ var _ = Describe("Networking", func() {
-				},
-			},
-		})
-		if err != nil {
-			Failf("unable to create test service named [%s] %v", svc.Name, err)
-		}
+	for _, svcSoak := range serviceSoakTests {
+		//copy to local to avoid range overwriting
+		timeoutSeconds := svcSoak.timeoutSeconds
+		serviceNum := svcSoak.service
+		It(fmt.Sprintf("should function for intrapod communication between all hosts in %v parallel services [Conformance]", serviceNum),
+			func() {
+				Logf("running service test with timeout = %v for %v", timeout, serviceNum)
+				runNetTest(timeoutSeconds, f, makePorts(serviceNum), nettestVersion)
+			})
+	}
+})
 
-		// Clean up service
-		defer func() {
-			By("Cleaning up the service")
-			if err = f.Client.Services(f.Namespace.Name).Delete(svc.Name); err != nil {
-				Failf("unable to delete svc %v: %v", svc.Name, err)
-			}
-		}()
+var pollings = 0
+
+//pollPeerStatus will either fail, pass, or continue polling.
+//returns true if polling succeeds.
+func pollPeerStatus(f *Framework, svc *api.Service, pollTimeoutSeconds time.Duration) bool {
+	Logf("Begin polling %v %v", svc.Name, pollings)
+	pollings += 1
+	getDetails := func() ([]byte, error) {
+		return f.Client.Get().
+			Namespace(f.Namespace.Name).
+			Prefix("proxy").
+			Resource("services").
+			Name(svc.Name).
+			Suffix("read").
+			DoRaw()
+	}
+
+	getStatus := func() ([]byte, error) {
+		return f.Client.Get().
+			Namespace(f.Namespace.Name).
+			Prefix("proxy").
+			Resource("services").
+			Name(svc.Name).
+			Suffix("status").
+			DoRaw()
+	}
 
-		By("Creating a webserver (pending) pod on each node")
+	passed := false
 
-		nodes, err := f.Client.Nodes().List(labels.Everything(), fields.Everything())
+	Logf("Begin polling " + svc.Name)
+	expectNoError(wait.Poll(2*time.Second, pollTimeoutSeconds, func() (bool, error) {
+		body, err := getStatus()
 		if err != nil {
 			Failf("Failed to list nodes: %v", err)
 		}
-		// previous tests may have cause failures of some nodes. Let's skip
-		// 'Not Ready' nodes, just in case (there is no need to fail the test).
-		filterNodes(nodes, func(node api.Node) bool {
-			return isNodeReadySetAsExpected(&node, true)
-		})
 
-		if len(nodes.Items) == 0 {
-			Failf("No Ready nodes found.")
-		}
-		if len(nodes.Items) == 1 {
-			// in general, the test requires two nodes. But for local development, often a one node cluster
-			// is created, for simplicity and speed. (see issue #10012). We permit one-node test
-			// only in some cases
-			if !providerIs("local") {
-				Failf(fmt.Sprintf("The test requires two Ready nodes on %s, but found just one.", testContext.Provider))
+		// Finally, we pass/fail the test based on if the container's response body, as to whether or not it was able to find peers.
+		switch {
+		case string(body) == "pass":
+			passed = true
+			return true, nil
+		case string(body) == "running":
+		case string(body) == "fail":
+			if body, err = getDetails(); err != nil {
+				Failf("Failed on attempt. Error reading details: %v", err)
```

Comment:
It may help with debugging to have the failure messages contain the service name, too.